File: hackathon-backend/src/main/java/hackathon/entity/ProjectAllocation.java
package hackathon.entity;

import javax.persistence.*;

@Entity
@Table(name = "project_allocation")
public class ProjectAllocation {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id = 0L;

    @ManyToOne
    @JoinColumn(name = "project_id")
    private Project project;

    @ManyToOne
    @JoinColumn(name = "employee_id")
    private Employee employee;

}

File: hackathon-backend-kotlin/src/main/kotlin/hackathon/backend/kotlin/entity/ProjectAllocation.kt
package hackathon.backend.kotlin.entity

import javax.persistence.*

@Entity
@Table(name = "project_allocation")
data class ProjectAllocation (
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long = 0L,

    @ManyToOne
    @JoinColumn(name = "project_id")
    var project: Project,

    @ManyToOne
    @JoinColumn(name = "employee_id")
    var employee: Employee
)
